--- tmux-3.5a.orig/compat.h	2025-06-16 18:04:22.000000000 +0000
+++ tmux-3.5a/compat.h	2025-06-16 18:29:05.000000000 +0000
@@ -471,4 +471,19 @@
 #define optreset           BSDoptreset
 #define optarg             BSDoptarg
 
+#ifdef __QNX__
+
+#include <unix.h>
+
+#ifndef IOV_MAX
+#include <sys/uio.h>
+#define IOV_MAX UIO_MAXIOV
+#endif
+
+#ifndef SA_RESTART
+#define SA_RESTART 0
+#endif
+
+#endif
+
 #endif /* COMPAT_H */
